tensorboard/webapp/runs/views/runs_table/runs_table_container.ts

```diff
@@ -389,8 +391,12 @@ export class RunsTableContainer implements OnInit, OnDestroy {
      * the store aware of the visibility of any run tables.
      */
     if (this.columns.includes(RunsTableColumn.CHECKBOX)) {
-      const runsExceedLimitForRoute$ = this.store.select(getRouteId).pipe(
+      const runsExceedLimitForRoute$ = this.store.select(getActiveRoute).pipe(
         takeUntil(this.ngUnsubscribe),
+        // BDTODO: This needs to be tested manually.
+        distinctUntilChanged((prevRoute, currRoute) => {
+          return areSameRouteAndExperiments(prevRoute, currRoute);
+        }),
```

Comment:
What (or what timing) makes the route might be different that we want to add `distinctUntilChanged` here?